### PR TITLE
Hdf5

### PR DESCRIPTION
**Pull request by bgruening**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bgruening** [comment]
@conda-forge-admin, please rerender

**Comment by lldelisle** [comment]
Many thanks, I hope it will solve the problem.

File: .ci_support/linux_mpimpichpython2.7.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ channel_targets:
 docker_image:
 - condaforge/linux-anvil-comp7
 hdf5:
-- 1.10.4
+- 1.10.5
```
Comment (ocefpaf) [inline .ci_support/linux_mpimpichpython2.7.yaml:12]: This version of `h5py` does not work with `hdf5 >1.10.4`. For a version that does work [you should use the dev release](https://github.com/conda-forge/h5py-feedstock/tree/dev) or wait for upstream to issue a new release.

Comment (bgruening) [reply .ci_support/linux_mpimpichpython2.7.yaml:12]: Thanks @ocefpaf ... thats too bad.

**Comment by bgruening** [comment]
@lldelisle this means you can use `h5py=2.10.0.dev` to get your stuff running for the moment.

File: .ci_support/linux_mpimpichpython2.7.yaml (hunk 1 above)
Comment (lldelisle) [reply .ci_support/linux_mpimpichpython2.7.yaml:12]: But on archlinux they managed see [here](https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/python-h5py&id=d1d4f225b9315ff4ed40f2be7bd6881e8a4d0197).

Comment (ocefpaf) [reply .ci_support/linux_mpimpichpython2.7.yaml:12]: You can port that patch. I went for a dev release b/c I needed more than just hdf5-1.10.5 support.

Comment (bgruening) [reply .ci_support/linux_mpimpichpython2.7.yaml:12]: That is the patch that they are using: https://github.com/h5py/h5py/commit/141eafa531c6c09a06efe6a694251a1eea84908d.patch

Comment (ocefpaf) [reply .ci_support/linux_mpimpichpython2.7.yaml:12]: That is only skipping a failing test if I got it right.
I remember the devs mentioning other issues, not sure if that is enough.

**Comment by tacaswell** [comment]
Sigh, "huh, I don't remember this, that sounds bad....what is this patch......oh, I wrote it...." :man_facepalming:

**Comment by ax3l** [comment]
Fixed in #59 :)

**Comment by minrk** [comment]
Merged as #59

**Pull request closed without merging**